### PR TITLE
Add script documentation.

### PR DESCRIPTION
**Pull request by Steve-O-Murphy**
Signed-off-by: Steve Murphy <stevemurphy@BCSR-11413.local>

### Description
I added documentation for these script APIs:
* Create script
* Delete script
* Execute script
* Get script contexts
* Get stored script.

I also added `index.md`.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: _opensearch/rest-api/_script-apis/delete-script.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+### Query parameters
+
+| Parameter | Data Type | Description | 
+:--- | :--- | :---
+| cluster_manager_timeout | Time | Amount of time to wait for a connection to the master node. Optional, defaults to `30s`. |
+| timeout | Time | The period of time to wait for a response. If a response is not received before the timeout value, the requ
+
+#### Sample request
+
+The following request deletes the `books` index.:
```
Comment (kolchfa-aws) [inline _opensearch/rest-api/_script-apis/delete-script.md:30]: This should be deleting the script, not the whole index.

File: _opensearch/rest-api/_script-apis/delete-script.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+````json
+DELETE /books
```
Comment (kolchfa-aws) [inline _opensearch/rest-api/_script-apis/delete-script.md:33]: DELETE _scripts/my-first-script

File: _opensearch/rest-api/_script-apis/delete-script.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+````
+
+#### Sample response
+
+The `DELETE /books` request returns the following field:
```
Comment (kolchfa-aws) [inline _opensearch/rest-api/_script-apis/delete-script.md:38]: This should be delete script

File: _opensearch/rest-api/_script-apis/delete-script.md [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+{
+  "acknowledged" : true
+}
+````
+
+To determine whether the stored script was successfully deleted, use the [Get stored script]({{site.url}}{{site.baseurl}}/opensearch/rest-api/_script-apis/get-stored-script/) API, passing the script name as the `script` path parameter.
+
+### Response fields
+
+The <HTTP METHOD> <endpoint> request returns the following response fields:
```
Comment (kolchfa-aws) [inline _opensearch/rest-api/_script-apis/delete-script.md:50]: Delete script request

File: _opensearch/rest-api/_script-apis/get-script-contexts.md [hunk 5]
```diff
@@ -0,0 +1,576 @@
...
+| Field | Data Type | Description | 
+:--- | :--- | :---
+| name | String | The context name. |
+|  methods | List | List of the context's allowable methods. See [Script object](#context-methods). |
+
+#### Context methods
+
+| Field | Data Type | Description | 
+:--- | :--- | :---
+| name | String | Mthod name. |
```
Comment (kolchfa-aws) [inline _opensearch/rest-api/_script-apis/get-script-contexts.md:567]: Method

File: _opensearch/rest-api/_script-apis/get-stored-script.md [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+
+````json
+GET _scripts/my-script
+````
+
+#### Sample response
+
+The `GET _scripts/my-script` request returns the following fields:
+
+````json
```
Comment (kolchfa-aws) [inline _opensearch/rest-api/_script-apis/get-stored-script.md:37]: Could we get an actual response from your my-first-script for consistency?

File: _opensearch/rest-api/_script-apis/index.md [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+redirect_from:
+  - /opensearch/rest-api/script-apis/
+---
+
+# Script APIs
+
+The script APIs allow you to work with stored scripts. Stored scripts are part of the cluster state and reduce compilation time and enhance search speed. The default scripting language is Painless. 
+
+You perform the following operations on stored scripts:
+* create
```
Comment (kolchfa-aws) [inline _opensearch/rest-api/_script-apis/index.md:16]: Can we make these links to pages?

Comment (Steve-O-Murphy) [reply _opensearch/rest-api/_script-apis/index.md:16]: @kolchfa-aws A list of links to pages is automatically generated.

File: _opensearch/rest-api/_script-apis/get-stored-script.md (hunk 6 above)
Comment (Steve-O-Murphy) [reply _opensearch/rest-api/_script-apis/get-stored-script.md:37]: I have made this change.

File: _opensearch/rest-api/_script-apis/index.md (hunk 7 above)
Comment (kolchfa-aws) [reply _opensearch/rest-api/_script-apis/index.md:16]: I see the above list rendering as plain text. I suggest we make those links to the pages they refer to.

File: _opensearch/rest-api/_script-apis/index.md [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+* update
+* execute
+* retrieve
+* delete
+ 
+You can also retrieve stored script contexts.
```
Comment (kolchfa-aws) [inline _opensearch/rest-api/_script-apis/index.md:22]: Make the last sentence a list item in the list above.

**Comment by ariamarble** [comment]
PR moved to #1094

**Comment by ariamarble** [comment]
Merged on #1094

File: _opensearch/rest-api/_script-apis/index.md (hunk 7 above)
Comment (Steve-O-Murphy) [reply _opensearch/rest-api/_script-apis/index.md:16]: @Naarcha-AWS, I looked at the production version and the links in the bulleted list (see attachment) are broken. They each include path segment `script-apis` that needs to be changed to `_script-apis`.

opensearch-script-link-errors

**Pull request closed without merging**